test/boost/sstable_test.cc

```diff
@@ -64,17 +64,25 @@ SEASTAR_TEST_CASE(composite_index) {
 }
 
 template<typename Func>
-inline auto
+inline future<>
 test_using_reusable_sst(schema_ptr s, sstring dir, sstables::generation_type::int_t gen, Func&& func) {
```

Comment:
Note for follow-up: it's better to use std::function instead of a template (or seastar::noncopyable_function). This reduces template bloat and documents the signature expected of func..

Reply:
Ah, this can't be done because func can return either void or future<>. Maybe after the series is done it's all void.

Reply:
Yes, I'll look into this in a separate PR.